scripts/code_coverage.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+set -e
+
+# Get member source directory and name from command line arguments
+if [ "$1" == "" -o "$2" == "" ]; then
+    echo "Command line argument member_crate_name or member_source_dir missing, Usage: ./codecoverage.sh member_crate_name member_source_dir"
+    exit 1
+fi
+member_crate_name=$1
+member_source_dir=$2
+source_root_dir="tari"
+build_dir="target/debug/"
+report_dir="report/"
+
+echo "Check if in correct directory":
+path=$(pwd)
+primary_dir=$(basename $path)
+if [ "$primary_dir" == "scripts" ]; then
+    echo "    + Moving to correct directory.."
+    cd ..
+fi
+
+# Check if in Tari primary folder before proceeding
+path=$(pwd)
+primary_dir=$(basename $path)
+if [ "$primary_dir" == "$source_root_dir" ]; then
+    echo "    + Correct directory"
+else
+    echo "    + Error: Incorrect directory -> start code_coverage from script or tari folder!"
+    exit 1
+fi
+
+echo "Check if grcov installed:"
+if [ "$(command -v grcov)" ]
+then
+    echo "    + Already installed"
+else
+    echo "    + Installing.."
+    cargo install grcov
+fi
+
+echo "Check if lcov installed:"
+if [ "$(command -v lcov)" ]
+then
+    echo "    + Already installed"
+else
+    echo "    + Installing.."
+    ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)" < /dev/null 2> /dev/null
```

Comment:
if ruby isn't installed this will fail here, won't it?

Reply:
Ruby seems to come preinstalled with Mac, might not work for other operating systems.